### PR TITLE
Feature | orders

### PR DESCRIPTION
**Pull request by WoLkErSs**

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class OrdersController < ApplicationController
-  def index; end
-
-  def show; end
+  def index
+    filter = OrdersQuery::FILTERS.values.include?(params[:filter]) ? params[:filter] : OrdersQuery::FILTERS[:default]
```
Comment (WoLkErSs) [inline app/controllers/orders_controller.rb]: extra code

Comment (za4emnik) [inline app/controllers/orders_controller.rb]: extra code

File: app/presenters/order_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class OrderPresenter
+  attr_reader :orders, :filter
+
+  def initialize(orders, filter)
+    @orders = orders
+    @filter = filter
+  end
+
+  def status(order)
+    order.in_progress? ? 'In progress' : order.status.capitalize
```
Comment (za4emnik) [inline app/presenters/order_presenter.rb]: https://apidock.com/rails/String/humanize

File: app/queries/orders_query.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class OrdersQuery
+  FILTERS = {
+    default: 'All orders',
```
Comment (za4emnik) [inline app/queries/orders_query.rb]: https://apidock.com/rails/String/humanize

File: app/queries/orders_query.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    canceled: 'Canceled'
+  }.freeze
+
+  def initialize(user)
+    @orders = user.orders
+  end
+
+  def orders_sort(filter)
+    case filter
+    when FILTERS[:in_progress] then @orders.where(status: 'in_progress')
```
Comment (za4emnik) [inline app/queries/orders_query.rb]:
```
return @orders.public_send(filter) if filter.include?

@orders
```

File: app/queries/orders_query.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class OrdersQuery
+  FILTERS = {
+    default: 'All orders',
+    in_progress: 'In progress',
+    processing: 'Processing',
```
Comment (za4emnik) [inline app/queries/orders_query.rb]: try to rethink logics

File: config/routes.rb [hunk 6]
```diff
@@ -10,7 +10,7 @@
   resources :books, only: %i[show index]
   resources :reviews, only: %i[create]
```
Comment (za4emnik) [inline config/routes.rb]: :

**Pull request merged**